File: 1593/1593.c
int splitAStringIntoTheMaxNumberOf(struct TreeNode* root) {
    if (!root) return 0;
    int l = splitAStringIntoTheMaxNumberOf(root->left);
    int r = splitAStringIntoTheMaxNumberOf(root->right);
    return 1 + (l > r ? l : r);
}

File: 1593/1593.cs
public class Solution {
    public int SplitAStringIntoTheMaxNumberOf(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(SplitAStringIntoTheMaxNumberOf(root.left), SplitAStringIntoTheMaxNumberOf(root.right));
    }
}